Fetch data from external source

erickrodrigs opened the pull request.

erickrodrigs committed:
add sheet key env var

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
add .env.sample

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
add axios

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
init api

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
create services to wrap spreadsheets requests

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
change api code to es6 syntax

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
add system metrics

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
create system controller

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
create module controller

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
create service controller

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
fetch system data from api

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
add cors to api

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
add @nuxtjs/axios

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
refactor system data service and mock its tests

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
fetch systems from api in home page

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
fetch module data from api and mock its tests

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
fetch service data from api and mock its tests

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
fix parse string numbers to integer

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
refactor system controller and metrics collector

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
refactor module controller and metrics collector

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
refactor service controller and metrics collector

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

jooaodanieel commented on api/index.js:
```diff
@@ -0,0 +1,22 @@
+
+app.use(cors())
+
+app.get('/systems', SystemController.index)
+
+app.get('/systems/:id', SystemController.find)
+
+app.get('/modules/:id', ModuleController.find)
+
+app.get('/services/:id', ServiceController.find)
```
Eu senti falta aqui de outras rotas
- `GET /systems/:id/modules`
- `GET /systems/:id/metrics`
- `GET /modules/:id/services`
- `GET /modules/:id/metrics`
- `GET /services/:id/metrics`

Com elas a gente pode ter um maior desacoplamento entre os dados requisitados. É um pouco mais custoso, porque exige maior quantidade de requests para obter os mesmos dados, mas acho que é importante para dar autonomia para os componentes.

jooaodanieel commented on pages/systems/_id.vue:
```diff
@@ -93,15 +93,14 @@ export default {
-      { source: 1, target: 3, type: 'sync' },
+      { source: 1, target: 3, type: 'async' },
+      { source: 3, target: 1, type: 'sync' },
       { source: 3, target: 2, type: 'async' },
-      { source: 3, target: 4, type: 'async' },
-      { source: 4, target: 5, type: 'async' },
+      { source: 3, target: 4, type: 'sync' },
+      { source: 4, target: 1, type: 'async' },
+      { source: 4, target: 2, type: 'async' },
+      { source: 5, target: 2, type: 'sync' },
       { source: 5, target: 4, type: 'sync' },
```
Entendi que esses dados são mocks posteriores à integração com as planilhas.
Eles ainda são necessários?

erickrodrigs committed:
add service communication parser

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
fix service page getting the same data

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
separate system routes

Now, we have the following routes for a specific system:
- GET systems/:id/modules, to get info about system's modules
- GET systems/:id/metrics, to get info about system's metrics

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
adapt system data service to obtain info from more than one route

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
add loading text and remove fake links at One System's page

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
add loading message at One Module Page

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
add loading message at One Service Page

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
fix system's modules links logic

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
refine system data service tests

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
separate module routes

Now, we have the following routes for a specific module:
- GET /modules/:id/services, to get info about module's services
- GET /modules/:id/metrics, to get info about module's metrics

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
adapt module data service to obtain info from more than one route

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
separate service routes

Now we have the following routes for a specific service:
- GET /services/:id/metrics, to get info about service's metrics

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
adapt service data api to obtain info from more than one route

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

jooaodanieel reviewed (changes requested):
Eu clonei local para executar e tudo funciona bem, a menos da visualização de serviço.
A tabela de operações assíncronas está vazia, mesmo para serviços que emitem eventos.
Dei uma inspecionada no código para ver se eu encontrava algo estranho, porém não achei a raiz do problema.

...

![localhost_3000_services_3(Laptop with HiDPI screen)](https://user-images.githubusercontent.com/12701604/116299135-af5b3580-a773-11eb-9ae6-bab98d9a6287.png)

erickrodrigs committed:
add label to service communication

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
add async operations to services

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
adapt service tests to the new async operation's format

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
display value on mouse over at Ruler component

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
add new metrics in a system view

now we have the following metrics:
- number of system's modules
- number of system's services
- percentage of synchronous coupling level
- percentage of asynchronous coupling level

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs commented:
@jooaodanieel eu tinha esquecido de pegar as operações assíncronas, agora acredito estar tudo ok.
Só queria comentar algumas coisas:

1. Acho que esse pull request tá ficando meio grande já hahahaha.
2. To meio perdido com relação a algumas métricas, acho que vale a pena eu conversar com vocês na próxima reunião.
3. Algumas novas métricas que a Thati passou não se encaixam no modelo que nosso componente de métricas aceita. Por exemplo, a métrica de Nome do serviço com maior quantidade de operações (nesse caso não teria um valor mínimo, máximo e etc). Seria legal pensar novamente numa nova estrutura para funcionar nesse caso (sem ideias por enquanto).

De todo modo, pedi um novo review. Qualquer detalhe estranho ou errado no que eu fiz, é só avisar.

erickrodrigs committed:
change metrics structure separating in nonMeasurable and measurable

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
refactor MetricsList and Ruler components

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
implement non-measurable metrics at system level

this commit also fixes some wrong implementations in measurable metrics.

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
refactor: separate each system's metric in a new file

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
add tests to validate system metrics

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>

erickrodrigs committed:
add new metrics: databases with most read/write connections

Signed-off-by: Erick Rodrigues <erick.rodrigues.santana14@gmail.com>
